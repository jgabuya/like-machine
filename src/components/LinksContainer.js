import React, { Component } from 'react';
import Link from './Link/Link';

import axios from '../util/axios-wrapper'

class LinksContainer extends Component {
    constructor(props) {
        super(props)

        this.state = {
            links: []
        }
    }

    async componentDidMount() {
        const result = await axios.get('/links');

        this.setState({
            links: result.data
        });
    }

    renderLinks() {
        return this.state.links.map((item, index) => {
            return (
                <Link
                    key={ index }
                    id={ item.id }
                    title={ item.title }
                    description={ item.description }
                    url={ item.url }
                    createdAt={ item.created_at }
                    liked={ item.liked }
                    likeCount={ item.like_count }
                    owned={ item.owned }
                    onLike={ () => {} }
                    onDelete={ () => {} }
                />
            )
        })
    }
    
    render() {
        return (
            <div>
                { this.renderLinks() }
            </div>
        );
    }
}

export default LinksContainer;
