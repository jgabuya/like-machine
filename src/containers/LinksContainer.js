import React, { Component } from "react";
import { Links } from "../components";

import { connect } from "react-redux";
import moment from "moment";
import fetchLinks from "../store/actions/fetch-links";
import deleteLink from "../store/actions/delete-link";
import likeLink from "../store/actions/like-link";
import unlikeLink from "../store/actions/unlike-link";

class LinksContainer extends Component {
	componentDidMount() {
        this.props.fetchLinks();        
    }
    
    groupLinksByDate(links) {
        return links.reduce((obj, curr) => {
            const key = moment(moment(curr.created_at).format('MMMM D, YYYY')).unix();

            if (typeof obj[key.toString()] === 'undefined') {
                obj[key.toString()] = [];
			}
            
            obj[key.toString()].push()

            return obj;
        }, {});
    }

	render() {
		return (
			<Links
				links={this.props.links}
                onDelete={this.props.onDelete}
                onLike={this.props.onLike}
                onUnlike={this.props.onUnlike}
			/>
		);
	}
}

const mapStateToProps = state => state;

const mapDispatchToProps = {
	fetchLinks: fetchLinks,
	onDelete: deleteLink,
    onLike: likeLink,
    onUnlike: unlikeLink
};

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(LinksContainer);

