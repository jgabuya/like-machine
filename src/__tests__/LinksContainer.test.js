import React from "react";
import LinksContainer from "../containers/LinksContainer";
import { shallow } from "enzyme";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";

describe("LinksContainer", () => {
    let wrapper;
    let store;
    const initialState = { links: [], user: null, formData: { url: "" } };

    beforeEach(() => {
        store = configureStore([thunk])(initialState);
        wrapper = shallow(<LinksContainer store={store} />);
    });

    it("renders without crashing", () => {
        expect(wrapper.length).toBe(1);
    });

    it("matches snapshot", () => {
        const tree = renderer.create(<LinksContainer store={store} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
