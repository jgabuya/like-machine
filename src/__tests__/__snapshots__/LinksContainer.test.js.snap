// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinksContainer matches snapshot 1`] = `<div />`;
